base/compiler/tfuncs.jl

```diff
@@ -48,27 +48,44 @@ end
 
 add_tfunc(throw, 1, 1, (@nospecialize(x)) -> Bottom, 0)
 
-# the inverse of typeof_tfunc
-# returns (type, isexact, isconcrete, istype)
-# if isexact is false, the actual runtime type may (will) be a subtype of t
-# if isconcrete is true, the actual runtime type is definitely concrete (unreachable if not valid as a typeof)
-# if istype is true, the actual runtime value will definitely be a type (e.g. this is false for Union{Type{Int}, Int})
-function instanceof_tfunc(@nospecialize(t))
+# works similarly to tuple, but with more type stability
+struct InstanceofResult
```

Comment:
Maybe could be an alias for `NamedTuple{(:type, :isexact, :isconcrete, :istype), Tuple{Type, Bool, Bool, Bool}}`?